docs/csharp/tour-of-csharp/classes-and-objects.md

```diff
@@ -0,0 +1,728 @@
+	}
+}
+public class Point3D: Point
+{
+	public int z;
+	public Point3D(int x, int y, int z) : 
+		base(x, y) 
+	{
+		this.z = z;
+	}
+}
+```
+
+A class inherits the members of its base class. Inheritance means that a class implicitly contains all members of its base class, except for the instance and static constructors, and the finalizers of the base class. A derived class can add new members to those it inherits, but it cannot remove the definition of an inherited member. In the previous example, `Point3D` inherits the `x` and `y` fields from `Point`, and every `Point3D` instance contains three fields, `x`, `y`, and `z`.
+
+An implicit conversion exists from a class type to any of its base class types. Therefore, a variable of a class type can reference an instance of that class or an instance of any derived class. For example, given the previous class declarations, a variable of type `Point` can reference either a `Point` or a `Point3D`:
+
+```csharp
+Point a = new Point(10, 20);
+Point b = new Point3D(10, 20, 30);
+```
+
+## Fields
+
+A ***field*** is a variable that is associated with a class or with an instance of a class.
+
+A field declared with the static modifier defines a static field. A static field identifies exactly one storage location. No matter how many instances of a class are created, there is only ever one copy of a static field.
+
+A field declared without the static modifier defines an instance field. Every instance of a class contains a separate copy of all the instance fields of that class.
+
+In the following example, each instance of the `Color` class has a separate copy of the `r`, `g`, and `b` instance fields, but there is only one copy of the `Black`, `White`, `Red`, `Green`, and `Blue` static fields:
+
+```csharp
+public class Color
+{
+	public static readonly Color Black = new Color(0, 0, 0);
+	public static readonly Color White = new Color(255, 255, 255);
+	public static readonly Color Red = new Color(255, 0, 0);
+	public static readonly Color Green = new Color(0, 255, 0);
+	public static readonly Color Blue = new Color(0, 0, 255);
+	private byte r, g, b;
+	public Color(byte r, byte g, byte b) 
+	{
+		this.r = r;
+		this.g = g;
+		this.b = b;
+	}
+}
+```
+
+As shown in the previous example, ***read-only fields*** may be declared with a `readonly` modifier. Assignment to a `readonly` field can only occur as part of the field’s declaration or in a constructor in the same class.
+
+## Methods
+
+A ***method*** is a member that implements a computation or action that can be performed by an object or class. ***Static methods*** are accessed through the class. ***Instance methods*** are accessed through instances of the class.
+
+Methods have a (possibly empty) list of ***parameters***, which represent values or variable references passed to the method, and a ***return type***, which specifies the type of the value computed and returned by the method. A method’s return type is void if it does not return a value.
```

Comment:
wouldn't be easier to say "Methods can have a list of **parameters**? and why are you using three asterisks instead of two?